test/Datadog.Trace.DuckTyping.Tests/ObscureObject.cs

```diff
@@ -0,0 +1,976 @@
+
+            private static DummyFieldObject PrivateStaticGetSelfType { get; } = DummyFieldObject.Default;
+
+            // *
+
+            public static DummyFieldObject PublicStaticGetSetSelfType { get; set; } = DummyFieldObject.Default;
+
+            internal static DummyFieldObject InternalStaticGetSetSelfType { get; set; } = DummyFieldObject.Default;
+
+            protected static DummyFieldObject ProtectedStaticGetSetSelfType { get; set; } = DummyFieldObject.Default;
+
+            private static DummyFieldObject PrivateStaticGetSetSelfType { get; set; } = DummyFieldObject.Default;
+
+            // *
+
+            public DummyFieldObject PublicGetSelfType { get; } = DummyFieldObject.Default;
+
+            internal DummyFieldObject InternalGetSelfType { get; } = DummyFieldObject.Default;
+
+            protected DummyFieldObject ProtectedGetSelfType { get; } = DummyFieldObject.Default;
+
+            private DummyFieldObject PrivateGetSelfType { get; } = DummyFieldObject.Default;
+
+            // *
+
+            public DummyFieldObject PublicGetSetSelfType { get; set; } = DummyFieldObject.Default;
+
+            internal DummyFieldObject InternalGetSetSelfType { get; set; } = DummyFieldObject.Default;
+
+            protected DummyFieldObject ProtectedGetSetSelfType { get; set; } = DummyFieldObject.Default;
+
+            private DummyFieldObject PrivateGetSetSelfType { get; set; } = DummyFieldObject.Default;
+
+            // ***
+
+            public static int? PublicStaticNullableInt { get; set; }
+
+            private static int? PrivateStaticNullableInt { get; set; }
+
+            public int? PublicNullableInt { get; set; }
+
+            private int? PrivateNullableInt { get; set; }
+
+            // ***
+
+            public int this[int index]
+            {
+                get => _dictioInt[index];
+                set => _dictioInt[index] = value;
+            }
+
+            public string this[string index]
+            {
+                get => _dictioString[index];
+                set => _dictioString[index] = value;
+            }
+
+            // ***
+
+            public TaskStatus Status { get; set; } = TaskStatus.RanToCompletion;
+
+            // *** Methods
+            public int Sum(int a, int b) => a + b;
+
+            public float Sum(float a, float b) => a + b;
+
+            public double Sum(double a, double b) => a + b;
+
+            public short Sum(short a, short b) => (short)(a + b);
+
+            private int InternalSum(int a, int b) => a + b;
+
+            public TestEnum ShowEnum(TestEnum val)
+            {
+                return val;
+            }
+
+            public T GetDefault<T>() => default;
+
+            public Tuple<T1, T2> Wrap<T1, T2>(T1 a, T2 b)
+            {
+                return new Tuple<T1, T2>(a, b);
+            }
+
+            public void Add(DummyFieldObject obj)
```

Comment:
This method looks unused in the Method tests

Reply:
I removed this method in https://github.com/DataDog/dd-trace-dotnet/pull/946/commits/e3020964dcbb2b80b94368777abbdd5b1de4e248, there are already some tests cases for duck chaining as a parameter method.